electron/main.ts

```diff
@@ -368,6 +370,30 @@ export const createWindow = (givenPath?: string) => {
 
   });
 
+  win.on('close', (e) => {
+    if (unsavedResourceCount === 0) {
+      return;
+    }
+
+    const message = unsavedResourceCount === 1
+      ? 'You have an unsaved resource'
```

Comment:
I wonder if unsaved files count makes more sense than unsaved resources count? You can have multiple unsaved resources in one file which doesn't give the user clear info on how many actions he has to perform before the warning disappears.

Example
- 2 unsaved files -> you close one and you know you still need to close another.
- 5 unsaved resources across 2 files -> you close one and don't know how many files you still need to find and save afterwards.

Reply:
Thanks for your contribution and good job, this looks great!
To answer you question, we only have the unsaved state for resources that are not already saved in a file. File changes are automatically saved.
Thus, using the unsaved resources count like you already did is the way to go.

Reply:
That makes sense, thanks for the clarification!